src/pages/JoinPage.tsx

```diff
@@ -1,31 +1,45 @@
 import { Button, Flex, VStack, Input, Progress, Text, FormControl } from "@chakra-ui/react";
 import Logo from "../components/Logo";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { FaArrowLeft } from "react-icons/fa";
 import { useNavigate } from "react-router-dom";
-import { useSocketJoinGameRoom } from "../hooks/useSocketJoinGameRoom";
 import { useCreatePlayer } from "../hooks/useCreatePlayer";
+import { useGetGame } from "../hooks/useGetGame";
 import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
 import { setPlayerName } from "../state/playerSlice";
-import { setGameCode, setHasJoinedGame } from "../state/gameSlice";
+import { setGameCode, setHasJoinedGame, setPollGetGameCount } from "../state/gameSlice";
 
 const CODE_LENGTH = 5;
 const MAX_NAME_LENGTH = 15;
 
 export default function JoinPage() {
   const [gameCode, setCode] = useState("");
   const [name, setName] = useState("");
-  const [displayForm, setDisplayForm] = useState("flex");
-  const [displayLoading, setDisplayLoading] = useState("none");
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
+
+  const { hasJoinedGame, gameStarted } = useAppSelector(state => state.game);
+  let pollGetGameCount = 0;
+  let pollStartGameTimeout: ReturnType<typeof setTimeout>;
+
   useCreatePlayer();
-  useSocketJoinGameRoom();
+  useGetGame();
 
-  const { hasJoinedGame } = useAppSelector(state => state.game);
+  useEffect(() => {
```

Comment:
We could technically move this useEffect, the next one, and the function `pollStartGame` into your `useGetGame` file (and maybe rename it into `usePollForGameStart`. That would make things nice and clean, and put all this related code into 1 place so its easy to see how it all links together (and easy to change in the future). But it looks like everything here is being handled very well, ya love to see it

Reply:
Made this change :)